lib/trento_web/controllers/v1/settings_controller.ex

```diff
@@ -93,4 +94,56 @@ defmodule TrentoWeb.V1.SettingsController do
       })
     end
   end
+
+  operation :update_activity_log_settings,
+    summary: "Updates the Activity Log settings",
+    tags: ["Platform"],
+    request_body:
+      {"ActivityLogSettings", "application/json", Schema.Platform.ActivityLogSettings},
+    responses: [
+      ok:
+        {"Activity Log settings saved successfully", "application/json",
+         Schema.Platform.ActivityLogSettings},
+      unprocessable_entity: Schema.UnprocessableEntity.response(),
+      not_found: Schema.NotFound.response()
+    ]
+
+  def update_activity_log_settings(%{body_params: body_params} = conn, _) do
+    %{retention_period: retention_period, retention_period_unit: retention_period_unit} =
+      body_params
+
+    case ActivityLog.change_retention_period(retention_period, retention_period_unit) do
```

Comment:
We could rely on a `with` and let the error bubble up
```elixir
with {:ok, updated_settings} <- ActivityLog.change_retention_period(retention_period, retention_period_unit) do
  render(conn, "activity_log_settings.json", %{
    activity_log_settings: updated_settings
  })
end
```

Reply:
![Screenshot from 2024-06-18 16-32-04](https://github.com/trento-project/web/assets/18139046/ecc0d556-bb21-4aee-b386-272baca5f48a)
Single clause `with` leads to a credo failure due to [this community readability default rule](https://hexdocs.pm/credo/Credo.Check.Readability.WithSingleClause.html) that is active.

Reply:
I guess you have a `with` + `else`, I was suggesting only a `with`.
We are anyway doing so elsewhere and credo is not complying :eyes:

Reply:
I see, good catch. Fixed this usage.